sdk/app_cpu1/common/sys/log.c

```diff
@@ -184,7 +184,7 @@ static void _do_log_to_stream(uint32_t elapsed_usec)
 
 void log_callback(void *arg)
 {
-    uint32_t elapsed_usec = scheduler_get_elapsed_usec();
+    uint32_t elapsed_usec = (uint32_t) scheduler_get_elapsed_usec();
```

Comment:
Hmm. does this work? truncating to integer? Seems like this should use the same `double` type as the scheduler?

Reply:
It seems to work just fine. I did this because the rest of the logging interface uses an abundance of `uint32_t` and it seemed more straightforward to add this cast vs change that whole interface to `double`. I was concerned that if the logging data format sends the timestamp, that format might expect a `uint32_t` instead of a `double`.

Reply:
Hmm. I think it probably works fine for nice integer numbers, like 10 kHz, 100usec, etc. But, I imagine for some "odd" stuff, this will cause issues.

Can you test with logging at some odd ratios? I think Anirudh is trying to use 88kHz PWM switching frequency with 22 kHz control rate, so ratio of 4x. Would be nice to confirm the system works with these settings. I bet with this integer casting here, it will skip some log samples sometimes

Reply:
@npetersen2 What's the final call on this? Can you re-review now that we've been using this code for logging on BP4?

Reply:
Let's keep it as is for now, but before `v1.3` release, we need to verify functionality.

See: https://github.com/Severson-Group/AMDC-Firmware/issues/404